### DISCUSSION TITLE
Dual license

### DISCUSSION THREAD

**Discussion by suprohub** [Pumpkin-MC/Pumpkin#476] [Ideas]
### I've searched existing issues and couldn't find a duplicate.

- [X] I confirm this is not a duplicate.

### Is your feature request related to a problem? Please describe.

I want to relicense Pumpkin under the "dual MIT / Apache-2.0 license". This allows users to select either license according to their own preferences. There are Very Good Reasons for this (see down).

1 The MIT license (arguably) requires binaries to reproduce countless copies of the same license boilerplate for every MIT library in use.
2 The Apache-2.0 license has protections from patent trolls and an explicit contribution licensing clause.
3 The Rust ecosystem is largely Apache-2.0. Being available under that license is good for interoperation and opens the doors to upstreaming pupmkin code into other projects.
4 The Apache license is incompatible with GPLv2, but MIT is compatible.

### Describe the solution you'd like

Relicense project

**Comment by suprohub** [comment]
@Snowiiii what do yoy think?

**Comment by rjc** [comment]
> 3 The Rust ecosystem is largely Apache-2.0. Being available under that license is good for interoperation and opens the doors to upstreaming pupmkin code into other projects.

That's not exactly what the [official Rust _Licenses_ page reads](https://www.rust-lang.org/policies/licenses):

> The Rust Programming Language and all other official projects, including this website, are generally dual-licensed:
>
>    [Apache License, Version 2.0](http://www.apache.org/licenses/LICENSE-2.0)
>    [MIT license](http://opensource.org/licenses/MIT)
>
> Specific licensing info for each project can be found in its GitHub Repository.

However, this is neither here nor there - software written in Rust (e.g. Pumpkin) doesn't have anything to do with Rust ecosystem _per se_ - you aren't contributing to Rust (the programming language).

If, however, you were merely thinking of other _3rd-party_ software **written** in Rust, then the _MIT_ license gives you a broader interoperability than Apache 2.0, anyway...

... s clearly shown in this counter-argument:

>  4 The Apache license is incompatible with GPLv2, but MIT is compatible.

Apache 2.0 is compatible with _GPLv3_, though.

Either way, KISS and use _MIT_ license :^)

**Comment by Snowiiii** [comment]
> The MIT license (arguably) requires binaries to reproduce countless copies of the same license 
boilerplate for every MIT library in use.

Sounds like you think the Compiler somehow packages the license inside the Compiler binary or something like that. Well this is not the case and licenses are also pretty important and not any kind of boilerplate
